src/main/java/org/apache/accumulo/testing/healthprobe/Monitor.java

```diff
@@ -0,0 +1,203 @@
+        Random tablet_index_generator = new Random();
+        Range range = pickRange(client.tableOperations(), opts.tableName, tablet_index_generator);
+        scanner.setRange(range);
+        if (opts.batch_size > 0) {
+          scanner.setBatchSize(opts.batch_size);
+        }
+        try {
+          long startTime = System.nanoTime();
+          int count = consume(scanner);
+          long stopTime = System.nanoTime();
+          MDC.put("StartTime", String.valueOf(startTime));
+          MDC.put("TabletIndex", String.valueOf(tablet_index_generator));
+          MDC.put("TableName", String.valueOf(opts.tableName));
+          MDC.put("TotalTime", String.valueOf((stopTime - startTime)));
+          MDC.put("StartRow", String.valueOf(range.getStartKey()));
+          MDC.put("EndRow", String.valueOf(range.getEndKey()));
+          MDC.put("TotalRecords", String.valueOf(count));
+
+          log.info("SCN starttime={} tabletindex={} tablename={} totaltime={} totalrecords={}", startTime,
+              tablet_index_generator, opts.tableName, (stopTime - startTime), count);
+          if (scannerSleepMs > 0) {
+            sleepUninterruptibly(scannerSleepMs, TimeUnit.MILLISECONDS);
+          }
+        } catch (Exception e) {
+          log.error(String.format(
+              "Exception while scanning range %s. Check the state of Accumulo for errors.", range), e);
+        }
+      }
+    }
+  }
+
+  public static int consume(Iterable<Entry<Key,Value>> iterable) {
+    Iterator<Entry<Key,Value>> itr = iterable.iterator();
+    int count = 0;
+    while (itr.hasNext()) {
+      Entry<Key,Value> e = itr.next();
+      Key key = e.getKey();
+      String readRow= key.getRow() + " " + key.getColumnFamily() + " " + key.getColumnQualifier() + " " + e.getValue();
+      count++;
+    }
+    return count;
+  }
+
+  public static long genLong(long min, long max, Random r) {
+    return ((r.nextLong() & 0x7fffffffffffffffL) % (max - min)) + min;
+  }
+
+  static byte[] genRow(long min, long max, Random r) {
+    return genRow(genLong(min, max, r));
+  }
+
+  public static byte[] genRow(long rowLong) {
+    return FastFormat.toZeroPaddedString(rowLong, 16, 16, EMPTY_BYTES);
+  }
+
+  public static Text genMaxRow(Text splitEnd) {
+    String padded = splitEnd.toString() + "0000000000000000".substring(splitEnd.toString().length());
+    Text maxRow = new Text(genRow( Long.parseLong(padded.trim(), 16) - 1l));
+    return maxRow;
+  }
+
+  public static Text genMinRow(Text splitStart) {
+    String padded = splitStart.toString()
+        + "0000000000000000".substring(splitStart.toString().length());
+    Text minRow = new Text(genRow(Long.parseLong(padded.trim(), 16)));
+    return minRow;
+  }
+
+  public static Range pickRange(TableOperations tops, String table, Random r)
+      throws TableNotFoundException, AccumuloSecurityException, AccumuloException {
+
+    ArrayList<Text> splits = Lists.newArrayList(tops.listSplits(table));
```

Comment:
@keith-turner @tushardhadiwal Could you cache the splits? Or maybe we could investigate doing that into Accumulo with an observable of some type...but a low hanging fruit could be to cache the splits since they won't change very frequently with something like a loadercache in guava that expires at a certain point. The splits themselves, even if they were to change, would still give us the same information here -- so perhaps a long lived cache would be idea?

Reply:
As a side node I'm not sure hyped about caching in Accumulo for this function -- so having a cache here would be awesome.

Reply:
@phrocker Sure. I have pushed the commit to cache the splits.